### PR TITLE
Properly handle input options

### PR DESCRIPTION
**Pull request by kick-the-bucket**
| Q             | A
| ------------- | ---
| Bug fix?      | Yes
| New feature?  | No
| BC breaks?    | No
| Deprecations? | No
| Fixed tickets | #1792 comment

### CODE REVIEW

**Comment by antonmedv** [comment]
Awesome!

**Comment by kick-the-bucket** [comment]
Looks like it's still handling the array options wrong in some edge cases, but I'll try to fix those as well.

**Comment by antonmedv** [comment]
@kick-the-bucket so? How it's going? Maybe delete `declare(strict_types=1);` if it helps?

**Comment by kick-the-bucket** [comment]
@rvitaliy I did find a way to pass `null` to both `VALUE_OPTIONAL` and `VALUE_IS_ARRAY | VALUE_OPTIONAL`.

> @kick-the-bucket so? How it's going? Maybe delete `declare(strict_types=1);` if it helps?
The option handling issues are logical, rather than type-related, so sadly this won't help.
Option handling might have never been fully correct (it certainly never worked with array options)...

**Comment by antonmedv** [comment]
So, everything looks ok to me. @kick-the-bucket @rvitaliy merging? 

And please update changelog by `php bin/changelog` command.

**Comment by kick-the-bucket** [comment]
Looks like all possible options values are now being handled correctly which was probably never the case since the feature was implemented almost 3 years ago :sweat_smile:

**Comment by rvitaliy** [comment]
LGTM 🍻

File: src/Executor/ParallelExecutor.php [hunk 1]
```diff
@@ -269,7 +268,9 @@ private function generateOptions(): string
             $inputs[] = Option::toString($this->input, $option);
         }
 
-        return implode(' ', $inputs);
+        return implode(' ', array_filter($inputs, function (string $item): bool {
```
Comment (rvitaliy) [inline src/Executor/ParallelExecutor.php]: in this case `Lambda function` can be `static`. [for more info see](http://php.net/manual/en/functions.anonymous.php#functions.anonymous-functions.static)

Comment (kick-the-bucket) [reply src/Executor/ParallelExecutor.php]: TIL!

**Pull request merged**